### PR TITLE
Boolean queries

### PR DESCRIPTION
**Pull request by ashwinkumar12345**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/elasticsearch/bool.md [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+}
+```
+
+For example, if you have all the works of Shakespeare indexed in an Elasticsearch cluster and you want to construct a single search query with the following requirements:
+
+1. The `text_entry` field must contain the word `love` and should contain either `life` or `grace`.
+2. The `speaker` field must not contain `ROMEO`.
+3. Filter these results to the play `Romeo and Juliet` without affecting the relevancy score.
+
+A sample `bool` query that meets these requirements is as follows:
```
Comment (aetter) [inline docs/elasticsearch/bool.md]: Use the following query:

File: docs/elasticsearch/bool.md [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+          "speaker": "PARIS",
+          "text_entry": "O love! O life! not life, but love in death!"
+        }
+      }
+    ]
+  }
+}
+```
+
+If you want to identify which of these clauses actually caused the matching results, name each of your queries with the `_name` parameter.
```
Comment (aetter) [inline docs/elasticsearch/bool.md]: name each query with the `_name` parameter.

File: docs/elasticsearch/bool.md [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+            "_name": "Romeo-and-Juliet-filter"
+          }
+        }
+      }
+    }
+  }
+}
+```
+
+You get back a `matched_queries` array that lists the queries that matched these results:
```
Comment (aetter) [inline docs/elasticsearch/bool.md]: Elasticsearch returns a `matched_queries` array

File: docs/elasticsearch/bool.md [hunk 4]
```diff
@@ -0,0 +1,206 @@
...
+```json
+"matched_queries": [
+  "Romeo-and-Juliet-filter",
+  "love-must",
+  "life-should"
+]
+```
+
+If you remove the queries not in this list, you will still see the exact same result.
+By examining which `should` clause matched, you can better understand the relevancy score of the results.
```
Comment (aetter) [inline docs/elasticsearch/bool.md]: Ending feels a little abrupt. Any additional detail we should add here, or this concludes it?

File: docs/elasticsearch/bool.md [hunk 5]
```diff
@@ -0,0 +1,206 @@
+---
+layout: default
+title: Boolean Queries
+parent: Elasticsearch
+nav_order: 11
+---
+
+# Boolean queries
+
+The `bool` query lets you combine multiple search queries with boolean logic. You can use boolean logic to include, exclude, or optionally include search queries.
```
[the remark this answers is not in the document]
Comment (ashwinkumar12345) [reply docs/elasticsearch/bool.md]: Include refers to "must" and optionally include to "should" and exclude to "not".  Maybe something like "You can use boolean logic for flexible inclusion and exclusion of search queries"?

File: docs/elasticsearch/bool.md (hunk 4 above)
Comment (ashwinkumar12345) [reply docs/elasticsearch/bool.md]: Yes, it does. I added an example for nested bool queries at the end.

File: docs/elasticsearch/bool.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (aetter) [reply docs/elasticsearch/bool.md]: Yep works for me.

**Pull request merged**